File: test/org/jetbrains/plugins/scala/lang/typeInference/TypeInferenceTest2.java
package org.jetbrains.plugins.scala.lang.typeInference;

import com.intellij.testFramework.ResolveTestCase;
import com.intellij.psi.PsiReference;
import com.intellij.psi.PsiElement;
import org.jetbrains.plugins.scala.util.TestUtils;
import org.jetbrains.plugins.scala.lang.psi.api.statements.params.ScParameter;
import org.jetbrains.plugins.scala.lang.resolve.ScalaResolveTestCase;

/**
 * @author ilyas
 */
public class TypeInferenceTest2 extends ScalaResolveTestCase {
  @Override
  protected String getTestDataPath() {
    return TestUtils.getTestDataPath() + "/typeInference/";
  }

  public void testFunParameter() throws Exception {
    PsiReference ref = configureByFile("expected/param/Param.scala");
    PsiElement resolved = ref.resolve();
    assertTrue(resolved instanceof ScParameter);
    String string = (((ScParameter) resolved)).calcType().toString();
    assertTrue(string.equals("String"));
  }
}

File: test/org/jetbrains/plugins/scala/lang/typeInference/TypeInferenceTest2.scala
package org.jetbrains.plugins.scala.lang.typeInference

import com.intellij.psi.PsiClass
import java.lang.String
import psi.api.statements.params.ScParameter
import psi.api.toplevel.typedef.ScTrait
import resolve.ScalaResolveTestCase
import util.TestUtils
import _root_.junit.framework.Assert._

/**
 * @author ilyas
 */

class TypeInferenceTest2 extends ScalaResolveTestCase {

  override protected def getTestDataPath: String = TestUtils.getTestDataPath() + "/typeInference/"

  def testFunParameter {
    val ref = configureByFile("expected/param/Param.scala")
    val resolved = ref.resolve
    assertTrue(resolved.isInstanceOf[ScParameter])
    val string = resolved.asInstanceOf[ScParameter].calcType.toString
    assertFalse(string == "Nothing")
  }

}